cmd/crank/beta/top/top.go

```diff
@@ -0,0 +1,287 @@
+  crossplane beta top
+
+  # Show resources utilization for all Crossplane pods in a specified namespace in a tabular format.
+  crossplane beta top -n <namespace>
+
+  # Show resources utilization for all Crossplane pods in the default 'crossplane-system' in a split format
+  crossplane beta top -s
+`
+}
+
+type topMetrics struct {
+	PodType      string
+	PodName      string
+	PodNamespace string
+	CPUUsage     int64
+	MemoryUsage  string
+}
+
+type defaultPrinterRow struct {
+	podType   string
+	namespace string
+	name      string
+	cpu       string
+	memory    string
+}
+
+func (r *defaultPrinterRow) String() string {
+	return strings.Join([]string{
+		r.podType,
+		r.namespace,
+		r.name,
+		r.cpu,
+		r.memory,
+	}, "\t") + "\t"
+}
+
+// Run runs the top command.
+func (c *Cmd) Run(k *kong.Context, logger logging.Logger) error { //nolint:gocyclo // TODO:(piotr1215) refactor to use dedicated functions
+	logger = logger.WithValues("cmd", "top")
+
+	var w = k.Stdout
+	tw := printers.GetNewTabWriter(w)
+	logger.Debug("Tabwriter header created")
+
+	// Build the config from the kubeconfig path
+	config, err := ctrl.GetConfig()
+	if err != nil {
+		return errors.Wrap(err, errKubeConfig)
+	}
+	logger.Debug("Found kubeconfig")
+
+	// Create the clientset for Kubernetes
+	k8sClientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return errors.Wrap(err, errCreateK8sClientset)
+	}
+	logger.Debug("Created clientset for Kubernetes")
+
+	// Create the clientset for Metrics
+	metricsClientset, err := versioned.NewForConfig(config)
+	if err != nil {
+		return errors.Wrap(err, errCreateMetricsClientset)
+	}
+	logger.Debug("Created clientset for Metrics")
+
+	ctx := context.Background()
+
+	pods, err := k8sClientset.CoreV1().Pods(c.Namespace).List(ctx, metav1.ListOptions{})
+
+	if err != nil {
+		return errors.Wrap(err, errFetchAllPods)
+	}
+
+	crossplanePods := getCrossplanePods(pods.Items)
+	logger.Debug("Fetched all Crossplane pods", "pods", crossplanePods, " from namespace", c.Namespace)
+
+	if len(crossplanePods) == 0 {
+		fmt.Println("No Crossplane pods found in the namespace", c.Namespace)
+		return nil
+	}
+
+	for i, pod := range crossplanePods {
+		podMetrics, err := metricsClientset.MetricsV1beta1().PodMetricses(pod.PodNamespace).Get(ctx, pod.PodName, metav1.GetOptions{})
+		if err != nil {
+			return errors.Wrap(err, errAddingPodMetrics)
+		}
+		var totalCPUUsage, totalMemoryUsage int64
+		for _, container := range podMetrics.Containers {
+			totalCPUUsage += container.Usage.Cpu().ScaledValue(resource.Milli)
+			totalMemoryUsage += container.Usage.Memory().ScaledValue(resource.Mega)
+		}
+
+		crossplanePods[i].CPUUsage = totalCPUUsage
+		crossplanePods[i].MemoryUsage = fmt.Sprintf("%dMi", totalMemoryUsage)
+	}
+
+	if err != nil {
+		return errors.Wrap(err, errGetPodMetrics)
+	}
+	logger.Debug("Added metrics to Crossplane pods")
+
+	if c.Split {
```

Comment:
before printing it would make sense to sort the pods, maybe by: type, namespace and name in this order, otherwise we risk flaking when watching, for example

Reply:
Good idea! The only caveat is that namespace will always be the same, so we can sort first by type and then by name.

Reply:
This is implemented.